targets/simple_switch/primitives.cpp

```diff
@@ -59,17 +60,16 @@ class modify_field_rng_uniform
     using hash = std::hash<std::thread::id>;
     static thread_local engine generator(hash()(std::this_thread::get_id()));
     using distrib64 = std::uniform_int_distribution<uint64_t>;
-    uint64_t lo = b.get_uint64();
-    uint64_t hi = e.get_uint64();
-    std::string warning = "";
+    auto lo = b.get_uint64();
+    auto hi = e.get_uint64();
     if (lo > hi) {
-        warning = " warning: result is not specified when lo > hi";
+        Logger::get()->warn("random result is not specified when lo > hi");
```

Comment:
you should exit from the primitive as this point, not construct the `distribution` object, because the constructor call has undefined behavior (and that undefined behavior could be a crash). I suggest one for the following:
 1. return without writing to the field
 2. write 0 to the field and return
 3. write a sentinel value to the field and return

Any of them is fine by me, maybe 1. is my favorite

Reply:
Commit pushed with changes for your choice 1.